scss/progressbar/_layout.scss

```diff
@@ -35,124 +38,101 @@
             .k-progress-status-wrap {}
         }
 
+
+        // Chunks
         > ul {
+            border-width: inherit;
+            border-style: inherit;
+            white-space: nowrap;
             display: flex;
+            align-items: stretch;
             flex: 1;
         }
         .k-item {
-            border-width: 0 transparent;
+            display: block;
+            border-width: 0;
             border-style: solid;
+            border-color: transparent;
         }
 
-
-        @include compact {
-            font-size: $spacer-y * .75;
-        }
-
-        > .k-reset {
-            list-style: none;
-            margin: 0;
-            padding: 0;
-            position: absolute;
-            left: -1px;
-            top: -1px;
-            width: 100%;
-            height: 100%;
-            border-radius: $border-radius;
-            white-space: nowrap;
-            border: inherit;
-        }
     }
 
+
+    // Horizontal
     .k-progressbar-horizontal {
         width: 27em;
-        height: $spacer-y * 2;
+        height: $progressbar-height;
         flex-direction: row;
 
+
         .k-progress-status-wrap {
             flex-direction: row;
             justify-content: flex-end;
             top: 0;
             left: 0;
         }
-
         > .k-state-selected {
-            left: -1px;
+            left: 0;
             right: auto;
-            top: -1px;
+            top: 0;
             height: 100%;
-            border-radius: $border-radius 0 0 $border-radius;
         }
 
+
+        // Chunks
         > ul {
             flex-direction: row;
         }
-        .k-item {
-            display: inline-block;
-            height: 100%;
-            border-style: solid;
-            margin-left: -1px;
-            border-width: 1px 0 1px 1px;
-
-            &.k-first {
-                margin-left: 0;
-            }
-
-            &.k-last {
-                border-right-width: 0;
-            }
-        }
+        .k-item {}
         .k-item + .k-item {
             border-left-width: 1px;
         }
 
 
+        // Horizontal reverse
         &.k-progressbar-reverse {
             flex-direction: row-reverse;
 
-            > .k-state-selected {
-                left: auto;
-                right: -1px;
-                border-radius: 0 $border-radius $border-radius 0;
-            }
-
-            .k-item {
-                border-width: 1px 0 1px 1px;
-            }
-
             .k-progress-status-wrap {
                 justify-content: flex-start;
                 left: auto;
                 right: 0;
             }
+            > .k-state-selected {
+                left: auto;
+                right: 0;
+            }
+
         }
     }
+
+
+    // Vertical
     .k-progressbar-vertical {
-        width: $spacer-x * 2;
+        width: $progressbar-height;
```

Comment:
I am using `$progressbar-height` to make it more consistent with Bootstrap. It's worth considering the establishment of common idioms / metaphors to work with.